src/realm/cluster.cpp

```diff
@@ -1590,8 +1606,9 @@ void Cluster::verify() const
         size_t col = spec.get_key(i).get_index().val + s_first_col_index;
```

Comment:
Would it be an idea to base this loop on iteration of colkeys instead of column indexes?

Reply:
It would. Will fix.